### PR TITLE
Update tests and fix node installation process.

### PR DESCRIPTION
**Pull request by seanlip**

### CODE REVIEW

**Comment by seanlip** [comment]
The main file to review is:
- scripts/run_karma_tests.sh

The other changes are due to tests actually failing on master due to previous modification of the preprocessCode function (mea culpa). We'll need to put in some presubmit checks in order to guard against this in the future.

File: scripts/run_karma_tests.sh [hunk 1]
```diff
@@ -25,18 +25,49 @@
...
+    else
+      NODE_FILE_NAME=node-v6.9.1-linux-x86
+    fi
+  fi
+
+  if [ ! -d "$TOOLS_DIR" ]; then
+    mkdir tools
+  fi
+
+  curl -o node-download.tgz http://nodejs.org/dist/v6.9.1/$NODE_FILE_NAME.tar.gz
```
Comment (rabidbit) [inline scripts/run_karma_tests.sh]: Why not download over https?

Comment (seanlip) [inline scripts/run_karma_tests.sh]: My bad (and good catch)! Done.

**Pull request merged**